backend/internal/database/setting.go

```diff
@@ -0,0 +1,161 @@
+	}
+
+	if !settingValue.Valid {
+		return settingId, "", nil
+	}
+	return settingId, settingValue.String, nil
+}
+
+// GetUserSetting returns the user-specific value for a particular
+// setting.  If there is no user-specific value stored for the given
+// user, then the default value is returned, if there is one.
+func GetUserSetting(redmineUserId int, settingName string) (string, error) {
+	settingId, settingValue, err := getSetting(settingName)
+	if err != nil {
+		return "", fmt.Errorf("getSetting() failed: %w", err)
+	}
+
+	sqlStmt := `
+		SELECT	value
+		FROM	user_setting
+		WHERE	redmine_user_id = ?
+		AND	setting_id = ?`
+
+	stmt, err := db.Prepare(sqlStmt)
+	if err != nil {
+		return "", fmt.Errorf("db.Prepare() failed: %w", err)
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.Query(redmineUserId, settingId)
+	if err != nil {
+		return "", fmt.Errorf("stmt.Query() failed: %w", err)
+	}
+	defer rows.Close()
+
+	userSettingFound := false
+	var userSettingValue sql.NullString
+
+	for rows.Next() {
+		err = rows.Scan(&userSettingValue)
+		if err != nil {
+			return "", fmt.Errorf("rows.Scan() failed: %w", err)
+		}
+
+		userSettingFound = true
+	}
+	err = rows.Err()
+	if err != nil {
+		return "", fmt.Errorf("rows.Next() failed: %w", err)
+	}
+
+	if !(userSettingFound && userSettingValue.Valid) {
+		// User setting was not found, or value is NULL.
+		// Return the default value instead.
+		return settingValue, nil
+	}
+
+	return userSettingValue.String, nil
+}
+
+// SetUserSetting assigns the user-specific value for a particular
+// setting.  If there is already a user-specific value stored for the
+// given setting, the new value replaces the old value.
+func SetUserSetting(redmineUserId int, settingName string, userSettingValue string) error {
+	settingId, _, err := getSetting(settingName)
+	if err != nil {
+		return fmt.Errorf("getSetting() failed: %w", err)
+	}
+
+	sqlStmt := `
+		INSERT INTO user_setting (redmine_user_id, setting_id, value)
+		VALUES (?, ?, ?)`
+
+	stmt, err := db.Prepare(sqlStmt)
+	if err != nil {
+		return fmt.Errorf("db.Prepare() failed: %w", err)
+	}
+	defer stmt.Close()
+
+	_, err = stmt.Exec(redmineUserId, settingId, userSettingValue)
+	if err != nil {
+		return fmt.Errorf("db.Exec() failed: %w", err)
+	}
+
+	return nil
+}
+
+// DeleteUserSetting removes the user-specific value for a particular
+// setting.
+func DeleteUserSetting(redmineUserId int, settingName string) error {
+	settingId, _, err := getSetting(settingName)
+	if err != nil {
+		return fmt.Errorf("getSetting() failed: %w", err)
+	}
+
+	sqlStmt := `
+		DELETE FROM user_setting
+		WHERE	redmine_user_id = ?
+		AND	setting_id = ?`
+
+	stmt, err := db.Prepare(sqlStmt)
+	if err != nil {
+		return fmt.Errorf("db.Prepare() failed: %w", err)
+	}
+	defer stmt.Close()
+
+	_, err = stmt.Exec(redmineUserId, settingId)
+	if err != nil {
+		return fmt.Errorf("db.Exec() failed: %w", err)
+	}
+
+	return nil
+}
```

Comment:
We assume for now that we will delete settings one by one. Bulk deletions might be possible in the future.